### ISSUE TITLE
functionality in setparts

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/partitions#5]
Given a set partition```c(1,1,1,3,2,2,1,1)``` it would be nice to turn it into a list of the sets it represents, in this case

```
> list(c(1,2,3,7,8),c(5,6),4)
[[1]]
[1] 1 2 3 7 8

[[2]]
[1] 5 6

[[3]]
[1] 4
```

Thus elements 1,2,3,7,8 are in set 1, elements 5,6 are in set 2, and element 4 in set 3

**Comment by RobinHankin** [comment]
Following works but needs checking for edge-cases:

```> x <- c(1,1,1,3,2,2,1,1)
> apply(outer(x,sort(unique(x)),`==`),2,which)
[[1]]
[1] 1 2 3 7 8

[[2]]
[1] 5 6

[[3]]
[1] 4
```

**Comment by RobinHankin** [comment]
Some edge-cases:
```
x <- c(1,2,3,3)

apply(outer(x,sort(unique(x)),`==`),2,which)
[[1]]
[1] 1

[[2]]
[1] 2

[[3]]
[1] 3 4
```

works fine; but:
```
x <- c(1,2,3)
apply(outer(x,sort(unique(x)),`==`),2,which)
[1] 1 2 3
> x <- c(1,1,2,2,3,3)
``` 
fails, and:
```
apply(outer(x,sort(unique(x)),`==`),2,which)
     [,1] [,2] [,3]
[1,]    1    3    5
[2,]    2    4    6
```

also fails.

**Comment by RobinHankin** [comment]
```
f <-
  function(x){
  jj <- sort(unique(x))
  M <- outer(jj,x,`==`)
  out <- lapply(split(M, seq_len(nrow(M))),which)
  names(out) <- jj
  return(out)
}
 f(c(1,2,3))
$`1`
[1] 1

$`2`
[1] 2

$`3`
[1] 3

 f(c(1,1,2,2,3,3))
$`1`
[1] 1 2

$`2`
[1] 3 4

$`3`
[1] 5 6
``

**Comment by RobinHankin** [comment]
I've just remembered that function ```listParts()``` does something similar

**Comment by RobinHankin** [comment]
but ```listParts()``` does not use set theory, and it should

**Comment by RobinHankin** [comment]
the branch fixed #6
